src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx

```diff
@@ -204,11 +220,29 @@ const CheckoutViewWrapper = () => {
   }, [isAuthenticated, user]);
 
   useEffect(() => {
-    if (promoCodeName) {
-      handleFetchPromotionCode(promoCodeName);
+    if (promoCodeName && currentSelectedPlan) {
+      handleFetchPromotionCode(currentSelectedPlan?.id, promoCodeName).catch((err) => {
+        const error = err as Error;
+        const errorMessage = error.message.includes('Promotion code with an id')
+          ? error.message
+          : 'Something went wrong, try again later';
+        setError('coupon', errorMessage);
+        errorService.reportError(error);
+        setPromoCodeData(undefined);
+      });
     }
   }, [promoCodeName]);
 
+  useEffect(() => {
+    if (state.error?.coupon || state.error?.auth || state.error?.stripe) {
```

Comment:
Add the conditionail in a variable with a descriptive name in order to understand better the reason